src/Microsoft.Health.SqlServer/Features/Storage/SqlServerSchemaDataStore.cs

```diff
@@ -117,6 +117,12 @@ public async Task<int> UpsertInstanceSchemaInformationAsync(string name, SchemaI
                 throw;
             }
 
+            if (sqlEx.IsCMKError())
+            {
```

Comment:
Will we still be able to tell that it failed because of a CMK dependency failure if client were to ask?

Reply:
Yes since this will always get captured by its caller. Here is a catch-all for CMK error in the schema worker https://github.com/microsoft/healthcare-shared-components/blob/ff171b29f6ca318dabede20694c1f19e02cfab36/src/Microsoft.Health.SqlServer/Features/Schema/SchemaJobWorker.cs#L94
I didn't want to log anything here so we avoid having a special case for CMK for each method